src/sqlfluff/rules/L003.py

```diff
@@ -552,7 +653,11 @@ def _process_current_line(
                 fixes = self._coerce_indent_to(
                     desired_indent=desired_indent,
                     current_indent_buffer=this_line["indent_buffer"],
-                    current_anchor=this_line["line_buffer"][0],
+                    current_anchor=[
+                        el
+                        for el in this_line["line_buffer"]
+                        if el not in this_line["indent_buffer"]
+                    ][0],
```

Comment:
The Anchor here was the start of the indent buffer, which was inconsistent with other anchors/triggers